relay/README.md

```diff
@@ -86,93 +64,144 @@ Scene 2:
+}
 ```
 
-After getting a stream to the relay node from its libp2p swarm,
-the dialing transport writes the header to the stream.
-The relaying node reads the header, gets a stream to the destination node,
-then writes the header to the destination stream and shortcircuits the two streams.
+### High level overview of establishing a relayed connection
+
+**Setup:**
+- Peers involved, A, B, R
+- A wants to connect to B, but needs to relay through R
+
+**Assumptions:**
+- A has connection to R, R has connection to B
+
+**Events:**
+- phase I: Open a request for a relayed stream (A to R).
+  - A dials a new stream `sAR` to R using protocol `/libp2p/circuit/relay/0.1.0`.
+  - A sends a CircuitRelay message with `{ type: 'HOP', srcPeer: '/p2p/QmA', dstPeer: '/p2p/QmB' }` to R through `sAR`.
+  - R receives stream `sAR` and reads the message from it.
+- phase II: Open a stream to be relayed (R to B).
+  - R opens a new stream `sRB` to B using protocol `/libp2p/circuit/relay/0.1.0`.
+  - R sends a CircuitRelay message with `{ type: 'STOP', srcPeer: '/p2p/QmA', dstPeer: '/p2p/QmB' }` on `sRB`.
+  - R sends a CircuitRelay message with `{ type: 'STATUS', code: 'OK' }` on `sAR`.
+- phase III: Streams are piped together, establishing a circuit
+  - B receives stream `sRB` and reads the message from it
+  - B sends a CircuitRelay message with `{ type: 'STATUS', code: 'OK' }` on `sRB`.
+  - B passes stream to `NewConnHandler` to be handled like any other new incoming connection.
 
-Each relayed connection corresponds to two multistreams,
-one between QmOne and QmRelay, the other between QmRelay and QmTwo.
+### Under the microscope
 
-Implementation details:
-- The relay node has the `Swarm.EnableRelaying` config option enabled
-- The relay node allows only one relayed connection between any two nodes.
-- The relay node validates the `src` header field.
-- The listening node validates the `dst` header field.
+- We've defined a max length for the multiaddrs of arbitrarily 1024 bytes
+- Multiaddrs are transfered on its binary packed format
+- Peer Ids are transfered on its non base encoded format (aka byte array containing the multihash of the Public Key).
 
 
-## Interfaces
+### Status codes table
 
-As explained above, the relay is both a transport (`tpt.Transport`)
-and a mounted stream protocol (`p2pnet.StreamHandler`).
-In addition it provides a means of specifying relay nodes to listen/dial through.
+This is a table of status codes and sample messages that may occur during a relay setup. Codes in the 200 range are returned by the relay node. Codes in the 300 range are returned by the destination node.
 
-TODO: the usage of p2pnet.StreamHandler is a little bit off, but it gets the point across.
+
+| Code  | Message                                           | Meaning    |
+| ----- |:--------------------------------------------------|:----------:|
+| 100   | OK                                                | Relay was setup correctly |
```

Comment:
Should this be "One leg of the relay was setup correctly", instead of the full relay?